src/test/java/org/prebid/server/bidder/admixer/AdmixerBidderTest.java

```diff
@@ -321,4 +370,23 @@ private static BidResponse givenBidResponse(Function<Bid.BidBuilder, Bid.BidBuil
     private static Map<String, JsonNode> givenCustomParams(String key, Object values) {
         return singletonMap(key, mapper.valueToTree(values));
     }
+
+    private static Imp givenImp(UnaryOperator<Imp.ImpBuilder> impCustomizer) {
+        return impCustomizer.apply(Imp.builder()
+                .id("123")
+                .ext(mapper.valueToTree(ExtPrebid.of(null,
+                ExtImpAdmixer.of("veryVeryVerySuperLongZoneIdValue", null,
+                        givenCustomParams("foo1", singletonList("bar1")))))))
+                .build();
+    }
+
+    //method where zoneId cut from ext and passed to tagId field
+    private static Imp givenImpWithParsedTagID(UnaryOperator<Imp.ImpBuilder> impCustomizer) {
```

Comment:
This method almost duplicates the above.
Can we call `givenImp(...)` here instead?

Reply:
The difference is that in givenImp tagId is located as a part of json in ext field, in givenImpWithParsedTagID tagId is cropped out of json and assigned to field tagId; givenImpWithParsedTagID is quite useful to compare imps after request(during request tagId is cropped and assigned)

Reply:
ok, but you can call `givenImp(...)` inside of `givenImpWithParsedTagID(...)`.
So, you'll rid off the code duplication.